### PR TITLE
Iteration 3 refactor

### PR DESCRIPTION
**Pull request by jrwhitmer**
In this PR we created a new module called findable that handles the repetitive code for finding objects in our classes by a specific attribute. We also pushed some functionality from sales analyst into our repository classes to help keep the code more organized. Lastly, we used some stubs for our tests for sales_analyst to cut back on rspec runtime.

### CODE REVIEW

File: spec/sales_analyst_spec.rb [hunk 1]
```diff
@@ -27,63 +27,69 @@
...
+      sales_analyst.stub(:merchant_ids_with_high_invoice_count).and_return([12334141, 12334146, 12334176, 12334183, 12334634, 12334942, 12335204, 12335213, 12335329, 12335417, 12336266, 12336430])
+
       expect(sales_analyst.top_merchants_by_invoice_count.class).to eq(Array)
       expect(sales_analyst.top_merchants_by_invoice_count.length).to eq(12)
       expect(sales_analyst.top_merchants_by_invoice_count.first.class).to eq(Merchant)
     end
 
-    xit 'can return which merchants are two standard deviations below the average' do
+    it 'can return which merchants are two standard deviations below the average' do
+      sales_analyst.stub(:merchant_ids_with_low_invoice_count).and_return([12334235, 12334601, 12335000, 12335560])
```
Comment (deebot10) [inline spec/sales_analyst_spec.rb:81]: These stubs are great for improving our performance. learning alot from this.

**Review by deebot10 — APPROVED** [review]

File: lib/findable.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+module Findable
+
```
[the remark this answers is not in the document]
Comment (ahenzler) [reply lib/findable.rb:2]: we need to remove the white space under module

File: lib/findable.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      instance.merchant_id == merchant_id
+    end
+  end
+
+  def find_all_by_invoice_id_repo(invoice_id, repo)
+    repo.find_all do |instance|
+      instance.invoice_id == invoice_id
+    end
+  end
+
```
[the remark this answers is not in the document]
Comment (ahenzler) [reply lib/findable.rb:26]: and whitespace here

File: lib/findable.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+module Findable
+
+  def find_by_id_repo(id, repo)
+    repo.find do |instance|
+      instance.id == id
+    end
+  end
```
Comment (deebot10) [inline lib/findable.rb:7]: This will quicken our run time

**Review by deebot10 — APPROVED** [review]

File: lib/merchants_repo.rb [hunk 5]
```diff
@@ -72,4 +72,10 @@ def delete(id)
     @merchants_list.delete(merchant)
   end
 
+  def merchant_id_array
+    @merchants_list.map do |merchant|
+      merchant.id
+    end
+  end
+
```
[the remark this answers is not in the document]
Comment (ahenzler) [reply lib/merchants_repo.rb:80]: whitespace

File: spec/transaction_repo_spec.rb [hunk 6]
```diff
@@ -58,9 +58,9 @@
...
       expect(expected.invoice_id).to eq(8)
-      expect(expected.credit_card_number).to eq(4242424242424242)
-      expect(expected.credit_card_expiration_date).to eq(220)
-      expect(expected.result).to eq("failed")
+      expect(expected.credit_card_number).to eq("4242424242424242")
+      expect(expected.credit_card_expiration_date).to eq("0220")
+      expect(expected.result).to eq(:failed)
     end
 
     it "can update transaction" do
```
Comment (ahenzler) [inline spec/transaction_repo_spec.rb:66]: great refactor overall. we need to remember to remove whitespace when we refactor.

**Review by ahenzler — APPROVED** [review]

**Review by rcasias — APPROVED** [review]

**Pull request merged**